types/src/transaction/transaction_header.rs

```diff
@@ -65,7 +65,15 @@ impl TransactionHeader {
     ) -> bool {
         match self {
             TransactionHeader::Deploy(header) => header
-                .is_valid(config, timestamp_leeway, at, transaction_hash)
+                .is_valid(
+                    config,
+                    timestamp_leeway,
+                    at,
+                    match transaction_hash {
+                        TransactionHash::Deploy(hash) => hash,
+                        _ => panic!("expected deploy hash"),
```

Comment:
Are we sure it's impossible to trigger this by sending malformed data? Because if it's possible, I think we should handle it gracefully (probably just return false here).

Reply:
True, that could still be an issue. Update in here: bc2a106837f19585d9366e805d94a23e96ae72d9